test/core/crawlers/basic_crawler.test.ts

```diff
@@ -577,6 +600,174 @@ describe.each(SingleStorageCase)('BasicCrawler - %s', (Emulator) => {
+        for (const args of warningSpy.mock.calls) {
+            expect(args.length).toBe(2);
+            expect(typeof args[0]).toBe('string');
+            expect(/Reclaiming failed request back to the list or queue/.test(args[0])).toBe(true);
+            expect(/requestHandler timed out after/.test(args[0])).toBe(true);
+            expect(/at Timeout\._onTimeout/.test(args[0])).toBe(false);
+            expect(args[1]).toBeDefined();
+        }
+
+        expect(errorSpy.mock.calls.length).toBe(1);
+        for (const args of errorSpy.mock.calls) {
+            expect(args.length).toBe(2);
+            expect(typeof args[0]).toBe('string');
+            expect(/Request failed and reached maximum retries/.test(args[0])).toBe(true);
+            expect(/requestHandler timed out after/.test(args[0])).toBe(true);
+            expect(/at Timeout\._onTimeout/.test(args[0])).toBe(false);
+            expect(args[1]).toBeDefined();
+        }
+    });
+
+    test('should log stack trace for non-timeout errors only when request will no longer be retried by default', async () => {
+        const sources = [{ url: `http://${HOSTNAME}:${port}` }];
+        const requestList = await RequestList.open(null, sources);
+
+        const crawler = new BasicCrawler({
+            requestList,
+            maxRequestRetries: 3,
+            requestHandler: () => {
+                throw new Error('Other non-timeout error');
+            },
+        });
+
+        // @ts-expect-error Overriding protected method
+        const warningSpy = jest.spyOn(crawler.log, 'warning');
+        // @ts-expect-error Overriding protected method
+        const errorSpy = jest.spyOn(crawler.log, 'error');
+
+        await crawler.run();
+
+        expect(warningSpy.mock.calls.length).toBe(3);
+        for (const args of warningSpy.mock.calls) {
+            expect(args.length).toBe(2);
+            expect(typeof args[0]).toBe('string');
+            expect(/Reclaiming failed request back to the list or queue/.test(args[0])).toBe(true);
+            expect(/Other non-timeout error/.test(args[0])).toBe(true);
+            expect(/at BasicCrawler\.requestHandler/.test(args[0])).toBe(false);
+            expect(args[1]).toBeDefined();
+        }
+
+        expect(errorSpy.mock.calls.length).toBe(1);
+        for (const args of errorSpy.mock.calls) {
+            expect(args.length).toBe(2);
+            expect(typeof args[0]).toBe('string');
+            expect(/Request failed and reached maximum retries/.test(args[0])).toBe(true);
+            expect(/Other non-timeout error/.test(args[0])).toBe(true);
+            expect(/at BasicCrawler\.requestHandler/.test(args[0])).toBe(true);
+            expect(args[1]).toBeDefined();
+        }
+    });
+
+    test('should log stack trace for timeout errors when verbose log is enabled', async () => {
+        log.setLevel(log.LEVELS.INFO);
+        process.env.CRAWLEE_VERBOSE_LOG = String(true);
```

Comment:
lol
```suggestion
        process.env.CRAWLEE_VERBOSE_LOG = 'true';
```

Reply:
Now I remember I did it once already 😄 it was late - my head was not working properly 🙃 jeez 😄